src/test/groovy/unit/org/pih/warehouse/receiving/ReceiptServiceSpec.groovy

```diff
@@ -0,0 +1,105 @@
+package unit.org.pih.warehouse.receiving
+
+import grails.testing.gorm.DataTest
+import grails.testing.services.ServiceUnitTest
+import org.pih.warehouse.api.PartialReceipt
+import org.pih.warehouse.core.ActivityCode
+import org.pih.warehouse.core.EventCode
+import org.pih.warehouse.core.Location
+import org.pih.warehouse.receiving.Receipt
+import org.pih.warehouse.receiving.ReceiptService
+import org.pih.warehouse.shipping.Shipment
+import org.pih.warehouse.shipping.ShipmentService
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Unroll
+
+@Unroll
+class ReceiptServiceSpec extends Specification implements ServiceUnitTest<ReceiptService>, DataTest {
```

Comment:
Great job writing tests!

It'd probably also be worth testing these scenarios:
1. partial receiving is enabled, it was partially received before, and now it's fully received
2. partial receiving is enabled, it was partially received before, and it still is (no events should happen)
3. partial receiving is enabled, it was fully received before, and it still is (no events should happen)